### PR TITLE
initial structure

### PR DESCRIPTION
**Pull request by VladaGromova**

### CODE REVIEW

File: src/main/java/com/epam/task/first/ArrayCreator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public List<Array> process(String filename) throws DataException {
+        List<Array> result = new ArrayList<Array>();
+        Array array = new Array();
+
+        List<String> stringsToValidate = dataReader.readData(filename);
+        for(String line : stringsToValidate){
+            if (validator.validate(line)){
+                array = parser.create(line);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/ArrayCreator.java:30]: Array array

File: src/main/java/com/epam/task/first/creator/ArrayParser.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Optional;
+
+public class ArrayParser {
+    // сюда передаются только валидные строчки
+    public Array create(String data){
+        List listOfNumbers = new ArrayList<Integer>();
+        int number;
+        String[] numbers = data.split(" ");
+        for(int i=0; i<numbers.length; ++i){
+            number = Integer.parseInt(numbers[i]);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/creator/ArrayParser.java:16]: int number;

File: src/main/java/com/epam/task/first/creator/ArrayParser.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ArrayParser {
+    // сюда передаются только валидные строчки
+    public Array create(String data){
+        List listOfNumbers = new ArrayList<Integer>();
+        int number;
+        String[] numbers = data.split(" ");
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/creator/ArrayParser.java:14]: parts

File: src/main/java/com/epam/task/first/creator/ArrayParser.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.epam.task.first.entities.Array;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ArrayParser {
+    // сюда передаются только валидные строчки
+    public Array create(String data){
+        List listOfNumbers = new ArrayList<Integer>();
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/creator/ArrayParser.java:12]: List<Integer>

File: src/main/java/com/epam/task/first/creator/ArrayValidator.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.epam.task.first.creator;
+
+import java.util.regex.Pattern;
+
+public class ArrayValidator {
+
+    public boolean validate(String line){
+        //boolean isValid = Pattern.matches("/^\\d+( \\d+)*$/", line);
+        boolean isValid = Pattern.matches("^[0-9 ]+$", line);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/creator/ArrayValidator.java:9]: private static final DIGIT_PATTERN = "^[0-9 ]+$";

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        return quantity;
+    }
+
+    public int getNumberOfNegativeNumbers(Array array){
+        List<Integer> elements = array.getElements();
+        checkValidity(elements);
+        int quantity = 0;
+        for (Integer element : elements){
+            if(element < 0){
+                ++quantity;
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:74]: quantity++;

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        return quantity;
+    }
+
+    public Array changeToTheOpposite(Array array){
+        List<Integer> elements = array.getElements();
+        checkValidity(elements);
+        int size = elements.size();
+        int newNumber;
+        for (int i=0; i< size; ++i){
+            newNumber = (elements.get(i))*(-1);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:86]: int newNumber

File: src/main/java/com/epam/task/first/logic/ArraySorter.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+public class ArraySorter {
+
+    private void checkValidity(List<Integer> elements){
+        if (elements.isEmpty()){
+            throw new IllegalArgumentException("Array is empty!");
+        }
+    }
+
+    private void swap(List<Integer> elements, int firstIndex, int secondIndex){
+        int tmp = elements.get(firstIndex);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArraySorter.java:16]: temp

File: src/main/java/com/epam/task/first/entities/Array.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.List;
+import java.util.Objects;
+
+public class Array {
+
+    private final List<Integer> elements;
+
+    public Array(Integer... elements){
+        this.elements = new ArrayList<Integer>();
+        this.elements.addAll(Arrays.asList(elements));
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/entities/Array.java:14]: this.elements  = Collections.unmodifableList(elements);

File: src/test/java/com/epam/task/first/data/DataReaderTest.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataReaderTest {
+
+    private  final static String TEST_DATA = "C:\\Java Web Development\\projects\\FirstTask\\src\\test\\resources\\input.txt";
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/first/data/DataReaderTest.java:11]: src/test/resources/input.txt

File: src/test/java/com/epam/task/first/data/DataReaderTest.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    private  final static String FALSE_DATA = "C:\\Java Web Development\\projects\\FirstTask\\src\\test\\resources\\output.txt";
+    private DataReader reader = new DataReader();
+
+    @Test
+    public void testReadDataShouldReadLinesFromFileWhenFileExists() throws DataException {
+        List<String> expected = new ArrayList<String>();
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/first/data/DataReaderTest.java:17]: Arrays.asList("1", "2");

File: src/test/java/com/epam/task/first/data/DataReaderTest.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        expected.add("1 2 3");
+        expected.add("4 5");
+        expected.add("7g 6 1111    9");
+        List<String> actual = reader.readData(TEST_DATA);
+        Assert.assertTrue(expected.equals(actual));
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/first/data/DataReaderTest.java:22]: Assert.assertEquals

File: src/test/java/com/epam/task/first/logic/ArrayCreatorTest.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+public class ArrayCreatorTest {
+
+    private final static List<String> TEST_DATA = Arrays.asList("1 2 3","3g 4 6");
+    static DataReader reader;
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/first/logic/ArrayCreatorTest.java:21]: private

**Review by FilippStankevichTR — CHANGES_REQUESTED** [review]

**Review by VladaGromova — COMMENTED** [review]
Thank you. I'll fix it.

**Pull request merged**